iceoryx_posh/test/moduletests/test_popo_base_server.cpp

```diff
@@ -0,0 +1,330 @@
+
+TEST_F(BaseServer_test, DestructorCallsDestroyOnUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "731fbc24-e4de-4223-af6b-baae6b87463d");
+
+    EXPECT_CALL(sut->port(), destroy).Times(1);
+
+    sut.reset(); // reset from the optional calls the dtor of the inner type
+}
+
+TEST_F(BaseServer_test, GetUidCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3ff821b6-4977-4405-b95d-60fb84933d28");
+
+    const UniquePortId uid;
+    EXPECT_CALL(sut->port(), getUniqueID).WillOnce(Return(uid));
+
+    EXPECT_THAT(sut->getUid(), Eq(uid));
+}
+
+TEST_F(BaseServer_test, GetServiceDescriptionCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "1481bfe5-4c70-4960-bb4c-92c637e2ec79");
+
+    EXPECT_CALL(sut->port(), getCaProServiceDescription).WillOnce(ReturnRef(sd));
+
+    EXPECT_THAT(sut->getServiceDescription(), Eq(sd));
+}
+
+TEST_F(BaseServer_test, OfferCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "5b50c652-4821-4fc2-abb4-d942db704894");
+
+    EXPECT_CALL(sut->port(), offer).Times(1);
+
+    sut->offer();
+}
+
+TEST_F(BaseServer_test, StopOfferCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "57e19e6e-a649-4e16-8cbc-7c7d922c0100");
+
+    EXPECT_CALL(sut->port(), stopOffer).Times(1);
+
+    sut->stopOffer();
+}
+
+TEST_F(BaseServer_test, IsOfferedCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b4b46a37-5331-4306-9df1-b092ea2d62ef");
+
+    constexpr bool IS_OFFERED{true};
+    EXPECT_CALL(sut->port(), isOffered).WillOnce(Return(IS_OFFERED));
+
+    EXPECT_THAT(sut->isOffered(), Eq(IS_OFFERED));
+}
+
+TEST_F(BaseServer_test, HasClientsCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "9c2eb6f3-5ce6-4bba-88de-08fcff87f5cf");
+
+    constexpr bool HAS_CLIENTS{true};
+    EXPECT_CALL(sut->port(), hasClients).WillOnce(Return(HAS_CLIENTS));
+
+    EXPECT_THAT(sut->hasClients(), Eq(HAS_CLIENTS));
+}
+
+TEST_F(BaseServer_test, HasRequestsCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "bcc738d3-21f7-4f36-9395-4a89f2f88a07");
+
+    constexpr bool HAS_REQUESTS{true};
+    EXPECT_CALL(sut->port(), hasNewRequests).WillOnce(Return(HAS_REQUESTS));
+
+    EXPECT_THAT(sut->hasRequests(), Eq(HAS_REQUESTS));
+}
+
+TEST_F(BaseServer_test, HasMissedRequestsCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "b0567e45-f66d-410b-b744-efdd4e566464");
+
+    constexpr bool HAS_MISSED_REQUESTS{true};
+    EXPECT_CALL(sut->port(), hasLostRequestsSinceLastCall).WillOnce(Return(HAS_MISSED_REQUESTS));
+
+    EXPECT_THAT(sut->hasMissedRequests(), Eq(HAS_MISSED_REQUESTS));
+}
+
+TEST_F(BaseServer_test, ReleaseQueuedRequestsCallsUnderlyingPort)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "3a226bd2-5c33-436a-adc6-d59e24de1074");
+
+    EXPECT_CALL(sut->port(), releaseQueuedRequests).Times(1);
+
+    sut->releaseQueuedRequests();
+}
+
+// BEGIN Listener and WaitSet related test
```

Comment:
The following tests contain a lot of duplicated code from `test_popo_base_client.cpp`. You could think about combining them in a `TYPED_TEST`.

Reply:
There are subtle differences and I thought the test are quite simple and didn't want to make them more complicated

Reply:
In another PR I made this a typed test to also test the derived `UntypedClient` and `Client`. With this change it would make it even more complex to introduce another dimension. I would keep it as is